spec/system/users_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe 'Users', type: :system do
+  it 'ユーザーはログインすると、/homeにリダイレクトされる' do
+    user = FactoryBot.create(:user)
+    visit new_user_session_path
+    fill_in 'Email', with: user.email
+    fill_in 'Password', with: user.password
+    click_button 'Log in'
+    expect(page).to have_current_path home_path
+  end
+
+  it '未ログインでusers/editを開いて、userログインして、users/editにリダイレクトされる' do
+    user = FactoryBot.create(:user)
+    visit edit_user_registration_path
+    fill_in 'Email', with: user.email
+    fill_in 'Password', with: user.password
+    click_button 'Log in'
+    expect(page).to have_current_path edit_user_registration_path
+  end
+
+  it 'ログインに失敗するとログイン画面のまま' do
+    user = FactoryBot.create(:user)
+    visit new_user_session_path
+    fill_in 'Email', with: user.email
+    fill_in 'Password', with: 'machigattapassword'
+    click_button 'Log in'
+    expect(page).to have_current_path new_user_session_path
+  end
+
+  it 'ユーザーはログインしてからadmins_rootに行ってもログインを求められる' do
+    user = FactoryBot.create(:user)
+    visit new_user_session_path
+    fill_in 'Email', with: user.email
+    fill_in 'Password', with: user.password
+    click_button 'Log in'
```

Comment:
管理側と同じく describe 等をうまく使って構造化した方がよいですがそれはそれとして、
テスト用のヘルパーを使うとこの処理は ↓ だけでよくなります。

```suggestion
    login_as(user, scope: :user)
```

参考: https://github.com/SonicGarden/sonic/blob/release-candidate/spec/support/devise.rb

Reply:
便利なのがあるんですね！
修正しました（ログイン操作を確認するところはそのまま）
[4fd0729](https://github.com/kk0000-kk/sakura-market/pull/19/commits/4fd072946d03c10620190176f50da172e18a33bb)